packages/spor-datepicker-react/src/TimePicker.tsx

```diff
@@ -0,0 +1,183 @@
+import { TimeValue } from "@react-types/datepicker";
+import { IconButton } from "@vygruppen/spor-button-react";
+import { createTexts, useTranslation } from "@vygruppen/spor-i18n-react";
+import {
+  DropdownLeftFill24Icon,
+  DropdownRightFill24Icon,
+} from "@vygruppen/spor-icon-react";
+import React from "react";
+import { useTimeFieldState } from "react-stately";
+import { StyledField } from "./StyledField";
+import { TimeField } from "./TimeField";
+import { getCurrentTime, useCurrentLocale } from "./utils";
+
+type TimePickerProps = Omit<BoxProps, "defaultValue"> & {
+  /** The label. Defaults to a localized version of "Time" */
+  label?: string;
+  /** The name of the form field, if used in a regular form */
+  name?: string;
+  /** The controlled value, if any.
+   *
+   * A `new Time(hours, minutes)` should be passed
+   **/
+  value?: TimeValue;
+  /** A default value, if any.
+   *
+   * A `new Time(hours, minutes)` should be passed. Defaults to the current time if not provided.
+   **/
+  defaultValue?: TimeValue;
+  /** Callback for when the time changes */
+  onChange?: (value: TimeValue) => void;
+  /** The maxiumum number of minutes to move when the step buttons are used.
+   *
+   * Defaults to 5 minutes.
+   *
+   * An example: If the time is at 13:37 and the stepGranularity is 15, clicking the step button forwards will move the time to 13:45. Next click will move it to 14:00.
+   */
+  stepGranularity?: number;
+  /** Whether or not the field is disabled */
+  isDisabled?: boolean;
+};
+/** A time picker component.
+ *
+ * This lets the user select a time of day, either through typing it in, using the up and down arrows to select the hour and minute, or by clicking the step buttons to move the time forwards or backwards in pre-defined increments.
+ *
+ * ```tsx
+ * <TimePicker />
+ * ```
+ *
+ * It can also be controlled:
+ *
+ * ```tsx
+ * <TimePicker value={new Time(13, 37)} onChange={setTime} />
+ * ```
+ *
+ * Note that the TimePicker uses the `Time` class to represent the time. This is a class that is part of the `@internationalized/date` package.
+ *
+ * @see https://spor.cloud.vy.no/komponenter/timepicker
+ */
+export const TimePicker = ({
+  label,
+  value,
+  defaultValue = getCurrentTime(),
+  onChange = () => {},
+  stepGranularity = 5,
+  isDisabled: isDisabledExternally = false,
+  name,
+  ...boxProps
+}: TimePickerProps) => {
+  const { isDisabled: isFormControlDisabled } = useFormControlContext() ?? {};
+  const isDisabled = isDisabledExternally ?? isFormControlDisabled ?? false;
+  const { t } = useTranslation();
+  const locale = useCurrentLocale();
+  const state = useTimeFieldState({
+    value,
+    defaultValue,
+    onChange,
+    locale,
+    isDisabled,
+  });
+
+  const dateTime = state.value as CalendarDateTime;
+
+  const handleBackwardsClick = () => {
+    const minutesToSubtract =
+      dateTime.minute % stepGranularity || stepGranularity;
```

Comment:
Whoah, matte!

Det jeg regner ut her (og i handleForwardClick) er hvor mange minutter man skal trekke fra (eller legge til), for å komme seg frem til nærmeste "hakk" i granularitetsstigen.

Dvs. at om man er på 13:37, med en `stepGranularity` på 5 minutter, og trykker på bakover-knappen, går man til 13:35. Trykker man en gang til går man til 13:30.

Sjeldent man får gleden av å bruke modulus-operatoren på helt ekte 😁

Det finnes faktisk en innebygget versjon av dette også – `state.value.cycle` – men den ruller deg ikke tilbake (eller frem) en time når man kommer til grensegangen mellom timene.